### PR TITLE
Workaround for self referenced attrbitues in pydantic

### PR DESCRIPTION
**Pull request by ilaner**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-5487

## Description
This replaces the `target` and `source` attributes in the relationship to the DB id, because Pydantic v1 doesn't support pickling self referenced models.
This supposed to be fixed in Pydantic v2.
In addition, it adds a `graph` option to the `prepare-content` command

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [ ] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

File: demisto_sdk/commands/content_graph/objects/relationship.py [hunk 1]
```diff
@@ -8,8 +8,10 @@
 
 class RelationshipData(BaseModel):
     relationship_type: RelationshipType
-    source: BaseContent
-    target: BaseContent
+        
+    # This are the database ids of the relationships
```
Comment (dantavori) [inline demisto_sdk/commands/content_graph/objects/relationship.py:12]:
```suggestion
    # These are the database ids of the relationships
```

File: demisto_sdk/commands/content_graph/interface/neo4j/neo4j_graph.py [hunk 2]
```diff
@@ -232,9 +232,9 @@ def _add_all_level_dependencies(
                     RelationshipType.DEPENDS_ON,
                     RelationshipData(
                         relationship_type=RelationshipType.DEPENDS_ON,
-                        source=obj,
+                        source_id=pack_id,
```
Comment (dantavori) [inline demisto_sdk/commands/content_graph/interface/neo4j/neo4j_graph.py:235]: is this the database id?

**Review by dantavori — COMMENTED** [review]
looks good to me,
1. is there a ticket for the bug? please add context to the PR
2. why is this a workaround? is there a plan to change it?
3. please add a test to make sure the bug was resolved correctly

File: demisto_sdk/commands/content_graph/interface/neo4j/neo4j_graph.py (hunk 2 above)
Comment (ilaner) [reply demisto_sdk/commands/content_graph/interface/neo4j/neo4j_graph.py:235]: Yes

**Comment by ilaner** [comment]
> looks good to me,
> 
> 1. is there a ticket for the bug? please add context to the PR
> 2. why is this a workaround? is there a plan to change it?
> 3. please add a test to make sure the bug was resolved correctly

1. Edited the description.
2. This supposed to be solved in Pydantic v2, we could discuss if we would like to change it once Pydantic v2 is released (should be by the end of Q1 2023).
3. Toggled the `USE_FUTURE` argument to `True` in the test to use multiprocessing in the test, to make sure it works properly.

File: demisto_sdk/commands/content_graph/objects/pack.py [hunk 3]
```diff
@@ -159,7 +160,7 @@ class Pack(BaseContent, PackMetadata, content_type=ContentType.PACK):  # type: i
     def validate_path(cls, v: Path) -> Path:
         if v.is_absolute():
             return v
-        return Path(get_content_path()) / v  # type: ignore
+        return Path(CONTENT_PATH) / v
```
Comment (dorschw) [inline demisto_sdk/commands/content_graph/objects/pack.py:163]:
```suggestion
        return Path(CONTENT_PATH, v)
```

File: demisto_sdk/commands/prepare_content/prepare_upload_manager.py [hunk 4]
```diff
@@ -25,18 +30,30 @@ def prepare_for_upload(
...
+                    raise ValueError(f"Could not find {input} in the graph.")
         if not output:
             if not input.is_dir():
                 input = input.parent
             output = input / content_item.normalize_name
         else:
             if output.is_dir():
                 output = output / content_item.normalize_name
-
+        output: Path  # Output is not optional anymore
```
Comment (dorschw) [inline demisto_sdk/commands/prepare_content/prepare_upload_manager.py:56]:
```suggestion
        output: Path  # silencing mypy: output is not optional from here on
```

File: demisto_sdk/commands/content_graph/objects/content_item.py [hunk 5]
```diff
@@ -43,7 +43,7 @@ class ContentItem(BaseContent):
     def validate_path(cls, v: Path) -> Path:
         if v.is_absolute():
             return v
-        return Path(get_content_path()) / v  # type: ignore
+        return Path(CONTENT_PATH) / v
```
Comment (dorschw) [inline demisto_sdk/commands/content_graph/objects/content_item.py:46]:
```suggestion
        return Path(CONTENT_PATH, v)
```

**Comment by ilaner** [comment]
Testing in content:
https://github.com/demisto/content/pull/24275

File: demisto_sdk/commands/content_graph/interface/graph.py [hunk 6]
```diff
@@ -118,6 +120,33 @@ def search(
...
+        Raises:
+            ValueError: If the path cannot be parsed locally
+            ValueError: If the path cannot be found in the graph
+
+        Returns:
+            Union[Pack, ContentItem]: The content item found
+        """
+        content_item = BaseContent.from_path(path)
+        if not isinstance(content_item, (ContentItem, Pack)):
+            raise ValueError(f"Could parse content_item from {path}")
```
Comment (dantavori) [inline demisto_sdk/commands/content_graph/interface/graph.py:141]:
```suggestion
            raise ValueError(f"Could not parse content_item from {path}")
```

**Review by dantavori — APPROVED** [review]

**Pull request merged**